mobile/status.go

```diff
@@ -0,0 +1,338 @@
+package status
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"unsafe"
+
+	"github.com/NaySoftware/go-fcm"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/status-im/status-go/api"
+	"github.com/status-im/status-go/logutils"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/profiling"
+	"github.com/status-im/status-go/services/personal"
+	"github.com/status-im/status-go/signal"
+	"github.com/status-im/status-go/transactions"
+	validator "gopkg.in/go-playground/validator.v9"
+)
+
+var statusBackend = api.NewStatusBackend()
+
+// All general log messages in this package should be routed through this logger.
+var logger = log.New("package", "status-go/lib")
+
+//GenerateConfig for status node
+func GenerateConfig(datadir string, networkID int) string {
+	config, err := params.NewNodeConfig(datadir, "", params.FleetBeta, uint64(networkID))
+	if err != nil {
+		return makeJSONResponse(err)
+	}
+
+	outBytes, err := json.Marshal(config)
+	if err != nil {
+		return makeJSONResponse(err)
+	}
+
+	return string(outBytes)
+}
+
+//StartNode - start Status node
+func StartNode(configJSON string) string {
+	config, err := params.LoadNodeConfig(configJSON)
+	if err != nil {
+		return makeJSONResponse(err)
+	}
+
+	if err := logutils.OverrideRootLog(true, "INFO", "", true); err != nil {
+		return makeJSONResponse(err)
+	}
+
+	api.RunAsync(func() error { return statusBackend.StartNode(config) })
```

Comment:
not really related to this PR. but it seems that RunAsync is totally unnecessary, every call is executed in its own thread. so, for example, StartNode can return an actual error, instead of using signals to report that error. and that means - part of the signals are not useful at all. and the other part of signals can be a simple a subscription over RPC.